File: updater/src/main/java/identifiers/Rasterizer2D.java
package identifiers;

import org.objectweb.asm.Type;
import utility.ClassWrapper;
import utility.DependsOn;
import utility.TypeUtilities;

@DependsOn(DualNode.class)
public class Rasterizer2D extends AbstractIdentifier {
    @Override
    public boolean isMatch(ClassWrapper classNode) {
        return classNode.isSuperClassEquals("DualNode") &&
                classNode.getInterfaces().isEmpty() &&
                classNode.getInstanceMethods().isEmpty() &&
                classNode.getInstanceFields().isEmpty() &&
                classNode.getCountStaticFieldsOfType(TypeUtilities.INT_ARRAY) == 1 &&
                classNode.getCountStaticFieldsOfType(Type.INT_TYPE) == 6;
    }
}

File: find-names/src/main/java/mapper/identifiers/classes/Rasterizer2D.kt
package mapper.identifiers.classes

import mapper.abstractclasses.IdentityMapper
import mapper.annotations.DependsOn
import mapper.predicateutilities.and
import mapper.predicateutilities.predicateOf
import mapper.predicateutilities.type
import mapper.wrappers.ClassWrapper
import org.objectweb.asm.Type

@DependsOn(DualNode::class)
class Rasterizer2D : IdentityMapper.Class() {
    override val predicate = predicateOf<ClassWrapper> { it.superType == type<DualNode>() }
            .and { it.interfaces.isEmpty() }
            .and { it.instanceFields.isEmpty() }
            .and { it.instanceMethods.isEmpty() }
            .and { it.staticFields.count { it.type == IntArray::class.type } == 1 }
            .and { it.staticFields.count { it.type == Type.INT_TYPE } == 6 }
}